### DISCUSSION TITLE
Vercel CLI fails with monorepos & pnpm

### DISCUSSION THREAD

**Discussion by yarinsa** [vercel/community#2106] [Help] [answered]
I want to trigger the deployment from out CI. 
This is my folder structure:
├── apps
   ├── storybook (SB)
   ├── console (Next.js)

I followed the instructions carefully and it seems like the CLI is intended to work from the[ root of the project. ](https://vercel.com/docs/concepts/deployments/configure-a-build#root-directory)

But when running 
`vercel build apps/console --debug` I can see the build logs online
```
ERR_PNPM_NO_MATCHING_VERSION_INSIDE_WORKSPACE  In : No matching version found for eslint-config@* inside the workspace
--
21:49:13.600 | Error: Command "pnpm install" exited with 1
21:49:14.402 | Deployment completed
21:49:13.997 | BUILD_UTILS_SPAWN_1: Command "pnpm install" exited with 1
``` 
I assume it's because I have in my package.json local package called "eslint-config": "workspace:*" and it doesn't have the workspace context.

I tried using the CLI help `vercel build --help' and it says there `  $ vercel build --cwd ./path-to-project`. But I am getting an install command as well. 

What are my options if I want to trigger it from GitHub and not vice-versa?

**Comment by amyegan** [comment, marked as answer]
It sounds like you already added a [Root Directory](https://vercel.com/docs/concepts/deployments/configure-a-build#root-directory) in your Project Settings. Do you get the same error if you leave the directory name out of the build command, simplifying it to `vercel build --debug`?

**Reply by yarinsa** [reply]
Then it doesn't find the project 
![Screenshot 2023-04-18 at 10 23 40](https://user-images.githubusercontent.com/44482362/232702397-1e709db9-97c8-4c39-ab9f-b5063213a09b.png)

**Reply by yarinsa** [reply]
@amyegan And if I remove the root directory settings then I am getting 
![Screenshot 2023-04-18 at 10 49 32](https://user-images.githubusercontent.com/44482362/232708933-2940a67a-2885-42ff-9578-1bc131304a2b.png)

**Reply by amyegan** [reply]
I believe you may need to keep the Root Directory setting, but leave the directory name out of the CLI build command. You should also double check that the project's package.json includes all of the necessary dependencies.

Various monorepo tools have their own setup requirements, so I can't offer specific advice. But you can find more setup and deployment information in our documentation and guides:
- [Monorepos](https://vercel.com/docs/concepts/monorepos)
- [Using Monorepos with Vercel CLI](https://vercel.com/docs/concepts/monorepos#using-monorepos-with-vercel-cli)
- [Deploying a Monorepo Using Yarn Workspaces to Vercel]( https://vercel.com/guides/deploying-yarn-monorepos-to-vercel)
